groups/stack/main.tf

```diff
@@ -0,0 +1,193 @@
+}
+
+# Configure the remote state data source to acquire configuration created
+# through the code in aws-common-infrastructure-terraform/groups/networking.
+data "terraform_remote_state" "networks_common_infra" {
+  backend = "s3"
+  config = {
+    bucket = var.aws_bucket
+    key    = "aws-common-infrastructure-terraform/common-${var.aws_region}/networking.tfstate"
+    region = var.aws_region
+  }
+}
+locals {
+  internal_cidrs = values(data.terraform_remote_state.networks_common_infra.outputs.internal_cidrs)
+  vpn_cidrs      = values(data.terraform_remote_state.networks_common_infra.outputs.vpn_cidrs)
+}
+
+# Remote state data source for Ireland, required for Concourse management CIDRs
+data "terraform_remote_state" "networks_common_infra_ireland" {
+  backend = "s3"
+  config = {
+    bucket = "development-eu-west-1.terraform-state.ch.gov.uk"
+    key    = "aws-common-infrastructure-terraform/common-eu-west-1/networking.tfstate"
+    region = "eu-west-1"
+  }
+}
+locals {
+  management_private_subnet_cidrs = values(data.terraform_remote_state.networks_common_infra_ireland.outputs.management_private_subnet_cidrs)
+}
+
+# Configure the remote state data source to acquire configuration
+# created through the code in the services-stack-configs stack in the
+# aws-common-infrastructure-terraform repo.
+data "terraform_remote_state" "services-stack-configs" {
+  backend = "s3"
+  config = {
+    bucket = var.aws_bucket # aws-common-infrastructure-terraform repo uses the same remote state bucket
+    key    = "aws-common-infrastructure-terraform/common-${var.aws_region}/services-stack-configs.tfstate"
+    region = var.aws_region
+  }
+}
+
+provider "vault" {
+  auth_login {
+    path = "auth/userpass/login/${var.vault_username}"
+    parameters = {
+      password = var.vault_password
+    }
+  }
+}
+
+data "vault_generic_secret" "secrets" {
+  path = "applications/${var.aws_profile}/${var.environment}/${local.stack_fullname}"
+}
+
+locals {
+  # stack name is hardcoded here in main.tf for this stack. It should not be overridden per env
+  stack_name       = "payments-service"
+  stack_fullname   = "${local.stack_name}-stack"
+  name_prefix      = "${local.stack_name}-${var.environment}"
+
+  public_lb_cidrs  = ["0.0.0.0/0"]
+  lb_subnet_ids    = "${var.admin_lb_internal ? local.application_ids : local.public_ids}" # place ALB in correct subnets
+  lb_access_cidrs  = "${var.admin_lb_internal ?
+                      concat(local.internal_cidrs,local.vpn_cidrs,local.management_private_subnet_cidrs,split(",",local.application_cidrs)) :
+                      local.public_lb_cidrs }"
+  app_access_cidrs = "${var.admin_lb_internal ?
+                      concat(local.internal_cidrs,local.vpn_cidrs,local.management_private_subnet_cidrs,split(",",local.application_cidrs)) :
+                      concat(local.internal_cidrs,local.vpn_cidrs,local.management_private_subnet_cidrs,split(",",local.application_cidrs),split(",",local.public_cidrs)) }"
+}
```

Comment:
Please consider splitting `data` and `locals` in to their own files, such as `data.tf` and `locals.tf` respectively to aid readability.

Reply:
done this bit here.